### PR TITLE
feat: improve dev setup

### PR DESCRIPTION
**Pull request by deryrahman**
Changes:
- [x] `make apply` using parameterize value of `DAGS_PATH`, and `OPTIMUS_SERVE_PORT`
- [x] provide dummy echo plugin as an example
- [x] add script to load secret defined in `setup.yaml`
- [x] script to install plugins through artifacts defined in `setup.yaml`

### CODE REVIEW

File: dev/Makefile [hunk 1]
```diff
@@ -2,34 +2,37 @@ EXPECTED_CONTEXT=colima
 NAMESPACE=optimus-dev
 OPTIMUS_ODPF_CHART=odpf/optimus
 
+DAGS_PATH?=./dags
```
Comment (smarch-int) [inline dev/Makefile:5]: using abspath is decent but only `$HOME/*` and `/tmp/colima/*` will work. Other paths won't work.

File: dev/plugins/optimus-plugin-echo.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+name: echo
+description: Echo sample plugin for optimus
+plugintype: task
+pluginversion: 0.1.0
```
Comment (smarch-int) [inline dev/plugins/optimus-plugin-echo.yaml:4]: image?

File: dev/Makefile [hunk 3]
```diff
@@ -2,34 +2,37 @@ EXPECTED_CONTEXT=colima
...
+OPTIMUS_PLUGINS_PATH?=./plugins
```
Comment (smarch-int) [inline dev/Makefile:6]: Here the assumption is to plugins are copied to this folder (extra step)

File: dev/load_secret.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+PROJECT=${PROJECT:-project-a}
+SECRET_NAME=${SECRET_NAME:-BQ_SERVICE_ACCOUNT}
```
Comment (smarch-int) [inline dev/load_secret.sh:4]: how to add multiple secrets ?

**Review by smarch-int — CHANGES_REQUESTED** [review]

File: dev/Makefile (hunk 1 above)
Comment (deryrahman) [reply dev/Makefile:5]: I made an assumption that user by default always create a folder inside `$HOME/*`, and any relative path will be expanded to absolute path on this loc https://github.com/odpf/optimus/blob/57725f11e1a216b7f8a3aa8baaff87a6c28946b4/dev/Makefile#L20

File: dev/load_secret.sh (hunk 4 above)
Comment (deryrahman) [reply dev/load_secret.sh]: added feat to load multiple secrets

File: dev/Makefile (hunk 3 above)
Comment (deryrahman) [reply dev/Makefile]: remove this one, use artifacts list inside setup.yaml instead

**Review by smarch-int — CHANGES_REQUESTED** [review]

File: dev/README.md [hunk 5]
```diff
@@ -1,34 +1,54 @@
...
++ `mkdir project-a`
++ `cd project-a`
++ `optimus init`
++ `optimus project register`
++ `optimus plugin sync`
++ then load the secret. [ref](#load-secrets)
+
+### Mounting the plugins
++ define plugin artifacts on `setup.yaml` under section `plugins`
++ pass the variable OPTIMUS_PLUGIN_ARTIFACTS when [applying optimus](#setup) (comma separated delimiter). eg.
```
Comment (smarch-int) [inline dev/README.md:37]: why to pass the var when we are listing the plugins in setup.yaml ?

Comment (deryrahman) [reply dev/README.md]: updated, committed the new one

**Review by smarch-int — APPROVED** [review]
LGTM!

**Pull request merged**